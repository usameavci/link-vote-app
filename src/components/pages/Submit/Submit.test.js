import React from "react";
import { act } from "react-dom/test-utils";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer, { act as actRenderer } from "react-test-renderer";

import PageSubmit from "./Submit";
import LinkContextProvider from "../../../contexts/LinkContext";

describe("Submit page component tests", () => {
  it("renders without crashing", async () => {
    const component = (
      <Router>
        <LinkContextProvider>
          <PageSubmit />
        </LinkContextProvider>
      </Router>
    );

    let getByTestId;

    await act(async () => ({ getByTestId } = await render(component)));

    expect(getByTestId("page-submit")).toBeInTheDocument();
    expect(getByTestId("submit-form")).toBeInTheDocument();
    expect(getByTestId("submit-form-input-title")).toBeInTheDocument();
    expect(getByTestId("submit-form-input-url")).toBeInTheDocument();
    expect(getByTestId("submit-form-button-submit")).toBeInTheDocument();
  });

  it("matches snapshots", async () => {
    const component = (
      <Router>
        <LinkContextProvider>
          <PageSubmit />
        </LinkContextProvider>
      </Router>
    );

    let tree;

    await actRenderer(async () => (tree = renderer.create(component)));

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
