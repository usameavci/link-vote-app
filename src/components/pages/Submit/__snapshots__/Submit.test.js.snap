// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submit page component tests matches snapshots 1`] = `
<div
  className="page-submit"
  data-testid="page-submit"
>
  <a
    className="page-submit-back"
    href="/"
    onClick={[Function]}
  >
    <i
      className="fa fa-arrow-left"
    />
    <span>
      Return to list
    </span>
  </a>
  <h1
    className="page-submit-title"
  >
    Add New Link
  </h1>
  <form
    className="page-submit-form"
    data-testid="submit-form"
    onSubmit={[Function]}
  >
    <div
      className="page-submit-group"
    >
      <label
        className="page-submit-label"
        htmlFor="title"
      >
        Link Name:
      </label>
      <input
        className="page-submit-input"
        data-testid="submit-form-input-title"
        onChange={[Function]}
        placeholder="e.g. Alphabet"
        type="text"
        value=""
      />
    </div>
    <div
      className="page-submit-group"
    >
      <label
        className="page-submit-label"
        htmlFor="title"
      >
        Link URL:
      </label>
      <input
        className="page-submit-input"
        data-testid="submit-form-input-url"
        onChange={[Function]}
        placeholder="e.g. http://abc.xyz"
        type="text"
        value=""
      />
    </div>
    <div
      className="text-right"
    >
      <button
        className="page-submit-button"
        data-testid="submit-form-button-submit"
        type="submit"
      >
        ADD
      </button>
    </div>
  </form>
</div>
`;
